src/lib/colors/colors.js

```diff
@@ -39,17 +39,16 @@ export default {
   normalize: (color = '', defaultColor = '0xffffffff') => {
     color = color.toString()
 
-    // its mappable, map it
-    if (colorsMap[color] !== undefined) {
-      return colorsMap[color]
-    }
-
-    // its a 0xRRGGBBAA color, store it and return it
+    // it is a 0xRRGGBBAA color, return it
     if (color.startsWith('0x') && color.length === 10) {
-      colorsMap[color] = color
       return color
     }
 
+    // it is a previously mapped color, return it
+    if (colorsMap[color] !== undefined) {
+      return colorsMap[color]
+    }
+
```

Comment:
this means you're no longer caching `0x` hex colors, is `(color.startsWith('0x') && color.length === 10)` faster then `(colorsMap[color] !== undefined)`?

Reply:
Correct, the difference between the 2 check is close to negligible. In the case of `0x` colors it doesn't really make sense to cache the result as the same value, without manipulation, will be returned anyway. So with this change we don't store data unnecessarily. For most normal app scenarios this won't be a problem, but when working with random colors in test scenarios the cache can fill up.

Reply:
Fair enough - thanks for explaining!